src/validation.cpp

```diff
@@ -923,7 +955,7 @@ bool AcceptToMemoryPoolWorker(CTxMemPool& pool, CValidationState& state, const C
         // invalid blocks (using TestBlockValidity), however allowing such
         // transactions into the mempool can be exploited as a DoS attack.
         unsigned int currentBlockScriptVerifyFlags = GetBlockScriptFlags(chainActive.Tip(), Params());
-        if (!CheckInputs(tx, state, view, true, currentBlockScriptVerifyFlags, true, true, txdata))
+        if (!CheckInputsFromMempoolAndCache(tx, state, view, pool, currentBlockScriptVerifyFlags, true, txdata))
```

Comment:
Perhaps even put the 2 calls to checkinputs in it, from https://github.com/bitcoin/bitcoin/pull/10192/commits/061d06faceed0272d1913800369768188876df77#diff-24efdb00bfbe56b140fb006b562cc70bR889